src/ngraph/ngraph_emitter.cc

```diff
@@ -1636,4 +1668,69 @@ void Emitter::CreateLossOps() {
   };
 }
 
+void Emitter::UnsupportedOps() {
+  for (auto kv : ngraph_op_funcs_) {
+    supported_ops[kv.first] = [](const NodePtr& node) { return true; };
+  }
+  supported_ops["BatchNorm"] = [](const NodePtr& node) {
+    bool out = true;
+    auto shape = TShape_to_NShape(node->inputs_[0]->shape_);
+    if (shape[1] % 8 != 0) {
+      // MXNet outperforms nGraph in this case.
+      out = false;
+    }
+    return out;
+  };
+  supported_ops["LeakyReLU"] = [](const NodePtr& node) {
+    bool out = true;
+    // We haven't yet implemented all activation functions for
+    // LeaklyReLU...
+    const std::string act_type =
+        get_default(node, "act_type", std::string("leaky"));
+    if (act_type != "leaky") {
+      out = false;
+    }
+    return out;
+  };
+  supported_ops["Deconvolution"] = [](const NodePtr& node) {
+    bool out = true;
+    auto num_group = get_default(node, "num_group", 1);
```

Comment:
NBD, but wasn't the plan, going forward, to use the per-operator, attribute-defining structs from the  `....-impl.h`files?

As opposed to repeating the logic for string-parsing and default-values like we're doing below?

Reply:
Generally, yes. I got lazy because these arguments don't have defaults, they have to be in the parameter dict.